kornia/geometry/ransac.py

```diff
@@ -132,39 +147,45 @@ def polish_model(self, kp1: torch.Tensor, kp2: torch.Tensor, inliers: torch.Tens
         )
         return model
 
+    def validate_inputs(self, kp1: Tensor, kp2: Tensor, weights: Optional[Tensor] = None) -> None:
+        if self.model_type in ['homography', 'fundamental']:
+            KORNIA_CHECK_SHAPE(kp1, ["N", "2"])
+            KORNIA_CHECK_SHAPE(kp2, ["N", "2"])
+            if not (kp1.shape[0] == kp2.shape[0]) or (kp1.shape[0] < self.minimal_sample_size):
+                raise ValueError(
+                    f"kp1 and kp2 should be \
+                                 equal shape at at least [{self.minimal_sample_size}, 2], \
+                                 got {kp1.shape}, {kp2.shape}"
+                )
+        if self.model_type == 'homography_from_linesegments':
+            KORNIA_CHECK_SHAPE(kp1, ["N", "2", "2"])
+            KORNIA_CHECK_SHAPE(kp2, ["N", "2", "2"])
+            if not (kp1.shape[0] == kp2.shape[0]) or (kp1.shape[0] < self.minimal_sample_size):
+                raise ValueError(
+                    f"kp1 and kp2 should be \
+                                 equal shape at at least [{self.minimal_sample_size}, 2, 2], \
+                                 got {kp1.shape}, {kp2.shape}"
+                )
+
     def forward(
-        self, kp1: torch.Tensor, kp2: torch.Tensor, weights: Optional[torch.Tensor] = None
-    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        self, kp1: Tensor, kp2: Tensor, weights: Optional[Tensor] = None
+    ) -> Tuple[Tensor, Tensor]:
         r"""Main forward method to execute the RANSAC algorithm.
 
         Args:
-            kp1 (torch.Tensor): source image keypoints :math:`(N, 2)`.
-            kp2 (torch.Tensor): distance image keypoints :math:`(N, 2)`.
-            weights (torch.Tensor): optional correspondences weights. Not used now
+            kp1 (Tensor): source image keypoints :math:`(N, 2)`.
```

Comment:
```suggestion
            kp1: source image keypoints :math:`(N, 2)`.
```
types are added automatically in the docs